lib/nrf_modem_lib/trace_mediums/Kconfig

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+choice NRF_MODEM_LIB_TRACE_MEDIUM
+	prompt "nRF modem trace medium"
```

Comment:
nit: just "Trace medium", both "nRF" and "modem" are redundant

Reply:
Applying the other suggestions, in `menuconfig` you'd get:
`Trace medium (UARTE1)` instead of
`nRF modem trace medium (Send Modem trace over UARTE1)`